iquip/apps/dataviewer.py

```diff
@@ -238,3 +240,169 @@ def __init__(self, parent: Optional[QWidget] = None):
+        """
+        TOTAL = 0
+        AVERAGE = 1
+        P1 = 2
+
+    dataTypeChanged = pyqtSignal(DataType)
+    thresholdChanged = pyqtSignal(int)
+
+    def __init__(self, parent: Optional[QWidget] = None):
+        """Extended."""
+        super().__init__(parent=parent)
+        layout = QGridLayout(self)
+        # first column (general information)
+        self.seriesLabel = QLabel("", self)
+        self.numberOfSamplesBox = QSpinBox(self)
+        self.numberOfSamplesBox.setMaximum(2**31 - 1)
+        self.numberOfSamplesBox.setButtonSymbols(QAbstractSpinBox.NoButtons)
+        self.numberOfSamplesBox.setReadOnly(True)
+        self.numberOfSamplesBox.setFrame(False)
+        self.thresholdBox = QSpinBox(self)
+        self.thresholdBox.setMaximum(2**31 - 1)
+        self.thresholdBox.setButtonSymbols(QAbstractSpinBox.PlusMinus)
+        self.thresholdBox.setPrefix("> ")
+        seriesLayout = QHBoxLayout()
+        seriesLayout.addWidget(QLabel("Series: "))
+        seriesLayout.addWidget(self.seriesLabel)
+        numberOfSamplesLayout = QHBoxLayout()
+        numberOfSamplesLayout.addWidget(QLabel("#Samples: "))
+        numberOfSamplesLayout.addWidget(self.numberOfSamplesBox)
+        thresholdLayout = QHBoxLayout()
+        thresholdLayout.addWidget(QLabel("Threshold: "))
+        thresholdLayout.addWidget(self.thresholdBox)
+        firstColumn = seriesLayout, numberOfSamplesLayout, thresholdLayout
+        for row, item in enumerate(firstColumn):
+            layout.addLayout(item, row, 0)
+        # second column (data type selection)
+        self.buttonGroup = QButtonGroup(self)
+        self.valueBoxes: Dict[DataPointWidget.DataType, QSpinBox] = {}
+        for dataType in DataPointWidget.DataType:
+            button = QRadioButton(dataType.name.capitalize(), self)
+            self.buttonGroup.addButton(button, id=dataType)
+            if dataType is DataPointWidget.DataType.TOTAL:
+                spinbox = QSpinBox(self)
+                spinbox.setMaximum(2**31 - 1)
+            else:
+                spinbox = QDoubleSpinBox(self)
+                if dataType is DataPointWidget.DataType.P1:
+                    spinbox.setMaximum(1)
+                else:
+                    spinbox.setMaximum(np.inf)
+            spinbox.setButtonSymbols(QAbstractSpinBox.NoButtons)
+            spinbox.setReadOnly(True)
+            spinbox.setFrame(False)
+            self.valueBoxes[dataType] = spinbox
+            layout.addWidget(button, dataType, 1)
+            layout.addWidget(QLabel(":"), dataType, 2)
+            layout.addWidget(spinbox, dataType, 3)
+        self.setDataType(DataPointWidget.DataType.P1)
+        # signal connection
+        self.buttonGroup.idToggled.connect(self._idToggledSlot)
+        self.thresholdBox.valueChanged.connect(self.thresholdChanged)
+
+    def seriesName(self) -> str:
+        """Returns the current data series name."""
+        return self.seriesLabel.text()
+
+    @pyqtSlot(str)
```

Comment:
Do you have any plan to use this method as a slot?

Reply:
Oh, I just marked the setters as slots, since Qt widgets are doing that usually.
Maybe because they can change the GUI, so it encourages to connect to signals and run on the main thread.
However, in PyQt, any functions can be connected to signals, so I don't mind if you want to remove those decorators.

Reply:
If it is a common way to define a setter, it looks better!
Please, never mind😃